src/flint/fmpq_poly.jl

```diff
@@ -260,14 +264,20 @@ end
 
 function shift_left(x::QQPolyRingElem, len::Int)
   len < 0 && throw(DomainError(len, "Shift must be non-negative"))
-  z = parent(x)()
+  return shift_left!(parent(x)(), x, len)
+end
```

Comment:
This method can completely be removed once https://github.com/Nemocas/AbstractAlgebra.jl/pull/2039 is available, as that generically delegates `shift_left` to `shift_left!`.